database/migration/src/main/resources/metadata_changesets/partition_metadata_table.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog objectQuotingStrategy="QUOTE_ALL_OBJECTS"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet id="update_indexes_metadata_table" author="cahrens" dbms="mysql,postgresql,hsqldb,mariadb">
+        <comment>Updates indexes on the METADATA_ENTRY table</comment>
+        <createIndex indexName="METADATA_ENTRY_UUID_KEY_CALL_INDEX_ATTEMPT_INDEX" tableName="METADATA_ENTRY">
+            <column name="WORKFLOW_EXECUTION_UUID"/>
+            <column name="CALL_FQN"/>
+            <column name="JOB_SCATTER_INDEX"/>
+            <column name="JOB_RETRY_ATTEMPT"/>
+            <column name="METADATA_KEY"/>
+        </createIndex>
+        <createIndex indexName="METADATA_ENTRY_UUID_KEY" tableName="METADATA_ENTRY">
+            <column name="WORKFLOW_EXECUTION_UUID"/>
+            <column name="METADATA_KEY"/>
+        </createIndex>
+        <dropIndex tableName="METADATA_ENTRY" indexName="METADATA_WORKFLOW_IDX"/>
+    </changeSet>
+    <changeSet id="compress_and_partition_metadata_table" author="cahrens" dbms="mysql">
+        <!-- Do not run this migration in Terra, where we will have already done the migration/table swap -->
+        <preConditions onFail="MARK_RAN">
```

Comment:
This precondition seems to only apply to partitioning/compressing? I suspect we will also want to create the indexes manually outside of liquibase, just not using percona, so I think we'll want the precondition on all of these changes

Reply:
We need to decide how we are doing the index update. If we do it outside of liquibase, I will add a precondition check to the update_indexes_metadata_table changeset above.